contracts/core/LensHub.sol

```diff
@@ -516,14 +516,21 @@ contract LensHub is ILensHub, LensNFTBase, VersionedInitializable, LensMultiStat
     /// ***************************************
 
     /// @inheritdoc ILensHub
-    function follow(uint256[] calldata profileIds, bytes[] calldata datas)
-        external
-        override
-        whenNotPaused
-    {
+    function follow(
+        uint256[] calldata profileIds,
+        uint256[] memory followedFromProfileIds,
+        bytes[] calldata datas
+    ) external override whenNotPaused {
+        for (uint256 i = 0; i < followedFromProfileIds.length; i++) {
```

Comment:
We should validate the input, checking `profileIds == followedFromProfileIds`.
Not a bad idea to validate the 3 arrays length at the beginning of the function imo.